language/transaction-builder/generator/src/lib.rs

```diff
@@ -35,7 +35,16 @@ pub fn read_abis<P: AsRef<std::path::Path>>(dir_path: P) -> anyhow::Result<Vec<S
         let mut buffer = Vec::new();
         let mut f = std::fs::File::open(path)?;
         f.read_to_end(&mut buffer)?;
-        abis.push(bcs::from_bytes(&buffer)?);
+        // Backwards compatibility -- first try to deserialize into the new `ScriptABI`, but if
+        // that fails, see if it's an old `ScriptABI` (which is now a `TransactionScriptABI`), and
+        // upgrade that into the new `ScriptABI`.
+        let deserialized_abi = match bcs::from_bytes::<ScriptABI>(&buffer) {
+            Err(_) => {
+                ScriptABI::TransactionScript(bcs::from_bytes::<TransactionScriptABI>(&buffer)?)
+            }
```

Comment:
While it might incidentally do the right thing here (for a known set of inputs), in general, retrying BCS deserialization with a different format is a really bad idea that can (and will) give unpredictable results.

Reply:
So, if we keep this logic, I would personally like to see a warning "TEMPORARY HACK" (to make sure people don't imitate this approach elsewhere) and a timeline to remove it.

Reply:
I've upgraded all of the old ABIs into the ABI format, and this upgrade logic is no longer needed.